core/state/statedb.go

```diff
@@ -1155,7 +1155,10 @@ func (s *StateDB) handleDestruction(nodes *trienode.MergedNodeSet) (map[common.H
 // trie, storage tries) will no longer be functional. A new state instance
 // must be created with new root and updated database for accessing post-
 // commit states.
-func (s *StateDB) Commit(deleteEmptyObjects bool) (common.Hash, error) {
+//
+// The associated block number of the state transition is also provided
+// for more chain context.
+func (s *StateDB) Commit(blockNumber uint64, deleteEmptyObjects bool) (common.Hash, error) {
```

Comment:
Nitpick, perhaps call it "block"? Being a uint64 is kind of self explanatory and less complex name